extensions/positron-python/python_files/positron/positron_ipykernel/positron_comm.py

```diff
@@ -87,7 +87,7 @@ class PositronComm:
 
     def __init__(self, comm: comm.base_comm.BaseComm) -> None:
         self.comm = comm
-        self.send_lock = threading.Lock()
+        self.send_lock = threading.RLock()
```

Comment:
This is the actual fix. The rest is to update the help tests to use the `PositronComm` instead of the base comm to be able to catch bugs like this one.